### PR TITLE
LPS-131941 url mapper to avoid rewrite.xml + LPS-130649 support locales in URLs

### PR DESCRIPTION
**Pull request by nhpatt**
Add a friendly URL mapper to return the root path and let the history router work without redirections. We are trying to avoid using url-rewrite to change URLs because Liferay Online won't let us customize it (we could hardcode it in the portal, as there are already old forums URLs hardcoded but ugly). The friendly URL mapper logic is very convoluted (and I'm not very smart) so I've tried a lot of approaches. Someone with previous knowledge I'm sure will know of a better solution.

Most promising one and works is: https://github.com/rotty3000/remote-web-component-poc injecting routes (and implementing Route & Router because are inside portal-impl) and setting the base path of the History/Browser router containing the friendly URL (so the basePath would be /ask/-/questions/). This works but means a third change of paths (old forums, questions and this one with the friendly URL. So this could work but I want to avoid it as it means reindexing everything again.

This PR works by cheating. I noticed that tags paths worked as expected (with and without parameters!) and it does not make any sense. What it happens is that they get picked by an obscure logic after the friendly URL mapper/translate part and they get transformed by the AssetTags URL mapper 🤦 , that checks with contains instead of equality, can not do anything with the URL and no parameters so it passes the base path to rendering and everything works.

So, I've done the same. It seems very simple but it isn't.

### CODE REVIEW

**Comment by nhpatt** [comment]
-- Forwarding and discussing later to avoid blocking the release

**Comment by nhpatt** [comment]
ci:forward

File: modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * @author Javier Gamarra
+ */
+@Component(
+	property = {
+		"com.liferay.portlet.friendly-url-routes=META-INF/friendly-url-routes/routes.xml",
+		"javax.portlet.name=" + QuestionsPortletKeys.QUESTIONS
+	},
+	service = FriendlyURLMapper.class
+)
+public class QuestionsFriendlyURLMapper extends DefaultFriendlyURLMapper {
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java:33]: I guess this way you override the portal url mapping and let React handle all the routing?

**Review by javierdearcos — APPROVED** [review]
Nice 😊

File: modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java (hunk 1 above)
Comment (nhpatt) [reply modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java:33]: Not exactly. An [URL mapper](https://help.liferay.com/hc/en-us/articles/360018150851-Adding-Friendly-URL-Mapping-to-the-Portlet) let you configure paths like `http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_journal_web_portlet_JournalPortlet&p_p_lifecycle=0&p_p_state=maximized&p_v_l_s_g_id=33726&p_p_auth=ZWt4266N` that have a lot of parameters and liferay stuff and clean it to something like `localhost:8080/guest/-/MAPPING/` and implicit parameters. 

But here we are changing the mode to contains so it acts differently, it matches the `questions/sectionTitle` URLs and returns nothing.

Comment (javierdearcos) [reply modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/portlet/route/QuestionsFriendlyURLMapper.java:33]: 🤔  Understood the first part. The second not so much...

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci exploded so retrying

**Pull request closed without merging**